arch/cortex-m/src/lib.rs

```diff
@@ -13,6 +13,255 @@ pub mod support;
 pub mod syscall;
 pub mod systick;
 
+#[cfg(all(target_arch = "arm", target_os = "none"))]
+#[inline(never)]
+unsafe fn kernel_hardfault_m3plus(faulting_stack: *mut u32) {
```

Comment:
```suggestion
unsafe fn kernel_hardfault_arm_v7m(faulting_stack: *mut u32) {
```

This is a name that better captures what cores can run this. (i.e. the new ARM M-23 cores are v8-m, which should use this just fine). The M0 is a v6-m.